examples/src/components/DrillWithExternalDataExample.jsx

```diff
@@ -0,0 +1,378 @@
+        }
+
+        return (
+            <div className="layout-wrapper">
+                {/* language=CSS */}
+                <style jsx>{`
+                    .layout-wrapper {
+                        display: grid;
+                        grid-template-columns: repeat(auto-fit, minmax(300px, 1fr));
+                        grid-column-gap: 20px;
+                        grid-row-gap: 20px;
+                        grid-template-rows: auto;
+                    }
+                    h3 {
+                        margin-top: 0;
+                        border-bottom: 2px solid #14b2e2;
+                        padding-bottom: 10px;
+                        margin-bottom: 10px;
+                    }
+                    .employeeTable {
+                        height: 100%;
+                        display: flex;
+                        flex-direction: column;
+                    }
+                    .employeeTableWrapper {
+                        position: relative;
+                        height: 100%;
+                        flex: 1 0 auto;
+                    }
+                    .details {
+                        padding-left: 20px;
+                    }
+                    .employeesHeading {
+                        margin-top: 20px;
+                    }
+                `}</style>
+                <div>
+                    <h3>States</h3>
+                    {stateTable}
+                    <h3>
+                        {!state && !location ? 'All Employees' : 'Employees from '}
+                        {state && (<span>
+                            {state.name}&nbsp;<button onClick={this.onClearState} className="button button-primary button-small button-icon-only icon-cross s-employee-heading-clear-state" />
+                        </span>)}
+                        {location && (<span>
+                            {location.name}&nbsp;<button onClick={this.onClearLocation} className="button button-primary button-small button-icon-only icon-cross s-employee-heading-clear-location" />
+                        </span>)}
+                    </h3>
+                    {employeeTable}
+                </div>
+                <div className="details" >
+                    <h3>
+                        Sales from {!state && !location && 'All Locations'}
+                        {state && (<span>
+                            {state.name}&nbsp;<button onClick={this.onClearState} className="button button-primary button-small button-icon-only icon-cross s-sales-heading-clear-state" />
+                        </span>)}
+                        {location && (<span>
+                            {location.name}&nbsp;<button onClick={this.onClearLocation} className="button button-primary button-small button-icon-only icon-cross s-sales-heading-clear-location" />
+                        </span>)}
+                    </h3>
+                    {totalSalesTable}
+                    {employee && (<div>
```

Comment:
`employee && renderEmployeeDetails(..)`

Reply:
This is actually two stage render. First when employee is selected, we render the title, but employeeDetails is null, then we render the details when the 3rd party call is resolved. So I will not move it to renderEmployeeDetails.